src/librustc_typeck/check/callee.rs

```diff
@@ -231,6 +231,10 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                             err.span_note(span, "defined here");
                         }
                     }
+                    if let Def::Local(_) = def {
+                        err.help("it's better to pick different names between variables and \
+                                  functions");
```

Comment:
I have some concerns about this message:

1. It's kind of judgemental. I personally don't consider it bad style to have a local variable with the same name as a function or method -- in fact, I consider it **good** style and I do it pretty regularly. For example, I might do `let labels = labels(...);` or `let labels = self.labels();`
2. The wording strongly suggests that there *is* a function with the same name as the local variable, but we don't check that this is the case. 
3. If we *did* do that check, then I would prefer to simplify highlight the local variable and the function and say something like "this local variable shadows the function, consider giving one of them a different name" -- i.e., just try to explain what's going wrong in an impartial way.